screensaver.fTVscreensaver/screensaver.py

```diff
@@ -0,0 +1,132 @@
+
+        self.get_images()
+        self.start_slideshow()
+
+    def calc_scrollspeed(self):
+        maxspeed = 100000
+        scrollspeed_delta = int(maxspeed / 100 * SCROLLSPEED)
+        speed = int(maxspeed - scrollspeed_delta + maxspeed / 10)
+        return str(speed)
+
+    def start_slideshow(self):
+        src_poster = self.images if SOURCE == '1' else self.artworks['poster']
+        src_fanart = self.images if SOURCE == '1' else self.artworks['fanart']
+
+        while not MONITOR.abortRequested() and not self.abortRequested:
+            for i in range(1,8):
+                winprop('fTVscreensaver.Poster.%s' % i, random.choice(src_poster))
+            for i in range(1,17):
+                winprop('fTVscreensaver.Fanart.%s' % i, random.choice(src_fanart))
+
+            MONITOR.waitForAbort(REFRESH_INTERVAL)
+
+    def get_images(self):
+        if ADDON.getSetting('source') == '1':
+            self.images = self.scan_folder(ADDON.getSetting('path'))
+
+        else:
+            poster = []
+            fanart = []
+
+            for media in ['Movies','TVShows']:
+                json_query = json_call('VideoLibrary.Get%s' % media,
+                                        properties=['art']
+                                        )
+
+                try:
+                    for item in json_query['result'][media.lower()]:
+                        if item['art'].get('poster'):
+                            poster.append(item['art'].get('poster'))
+                        if item['art'].get('fanart'):
+                            fanart.append(item['art'].get('fanart'))
+                except Exception:
+                    pass
+
+            self.artworks = {}
+            self.artworks['poster'] = poster
+            self.artworks['fanart'] = fanart
+
+    def scan_folder(self, path):
+        dirs, files = xbmcvfs.listdir(path)
+
+        images = [
+            xbmc.validatePath(path + f) for f in files
+            if f.lower()[-3:] in ('jpg', 'png')
+        ]
+
+        if RECURSIVE:
+            for directory in dirs:
+                if directory.startswith('.'):
+                    continue
+
+                images.extend(
+                    self.scan_folder(
+                        xbmc.validatePath('/'.join((path, directory, '')))
+                    )
+                )
+
+        return images
+
+    def exit(self):
+        self.abortRequested = True
+        for i in range(1,8):
+            winprop('fTVscreensaver.Poster.%s' % i, clear=True)
+        for i in range(1,17):
+            winprop('fTVscreensaver.Fanart.%s' % i, clear=True)
+        self.close()
+
+
+if __name__ == '__main__':
+    screensaver = Screensaver('screensaver.fTVscreensaver.xml',ADDON_PATH,'default')
+    screensaver.doModal()
+    del screensaver
+    sys.modules.clear()
```

Comment:
any specific reason to clear the sys.modules when finishing the addon?

Reply:
To be honest, It's a leftover from back in the days where I had no clue about Python and I just made this script based on another one.
But yeah, makes no sense. I'll remove it.